File: python/06_arrays.py
# 06 lists / arrays

nums = [1, 2, 3, 4, 5]
print(nums)
print(nums[0])       # first
print(nums[-1])      # last

nums.append(6)
print(nums)

# list comprehension
squares = [x * x for x in nums]
print(squares)

# slice
print(nums[1:4])

File: rust/06_arrays.rs
fn main() {
    let mut nums = vec![1, 2, 3, 4, 5];
    println!("{:?}", nums);
    println!("{}", nums[0]);
    println!("{}", nums[nums.len() - 1]);

    nums.push(6);
    println!("{:?}", nums);

    let squares: Vec<i32> = nums.iter().map(|x| x * x).collect();
    println!("{:?}", squares);
}
